### ISSUE TITLE
Getting unlexable error when parsing psql meta-commands

### ISSUE THREAD

**Issue by EdmundsEcho** [sqlfluff/sqlfluff#1327]
Here is the minimal SQL example which reproduces the issue

The linter bugs out anytime the code includes psql meta-commands

```sql
begin;

\echo 'Creating the api schema...'  -- <<< the offending code

CREATE schema authorization api;

comment on schema api is
    'Schema that defines an API suitable to be exposed through PostgREST';

set role api;
```

## Expected Behaviour
It should ignore where it does not have a point of view.

## Error returned
```fish
FAIL - Zero length Lex item returned from '<unlexable>'. Report this as a bug.
```

## Steps to Reproduce
sqlfluff lint file_with_meta_commands.sql

## Dialect
postgres

## Version
Include the output of `sqlfluff --version` along with your Python version
fluff: 0.6.4
pip: 3.9.0

## Configuration
no configuration
